pkg/controller/controller_clusterservicebroker.go

```diff
@@ -543,24 +543,35 @@ func (c *controller) reconcileClusterServicePlanFromClusterServiceBrokerCatalog(
 	servicePlan.Spec.ClusterServiceBrokerName = broker.Name
 
 	if existingServicePlan == nil {
-		otherServicePlan, err := c.clusterServicePlanLister.Get(servicePlan.Name)
+		//otherServicePlan, err := c.clusterServicePlanLister.Get(servicePlan.Name)
+
+		selector := labels.NewSelector()
+		requirement, err := labels.NewRequirement("Spec.ExternalID", "==", []string{servicePlan.Spec.ExternalID})
 		if err != nil {
-			// we expect _not_ to find a service class this way, so a not-
-			// found error is expected and legitimate.
-			if !errors.IsNotFound(err) {
-				return err
-			}
-		} else {
+			errMsg := fmt.Sprintf("Error building requirement for reconciling class %q from broker %q: %s",
+				pretty.ClusterServicePlanName(servicePlan), broker.Name, err.Error(),
+			)
+			glog.Error(pcb.Message(errMsg))
+			return fmt.Errorf(errMsg)
+		}
+		selector.Add(*requirement)
+		otherServicePlans, err := c.clusterServicePlanLister.List(selector)
+		if err != nil {
+			return err
+		}
+		if len(otherServicePlans) > 0 {
```

Comment:
golang tip... I believe this "if" is not needed since the "for" loop below will do nothing if its zero length.  Same for previous check like this above.